### ISSUE TITLE
Missing check for STATIC method when executing

### ISSUE THREAD

**Issue by ilovejpf** [javapathfinder/jpf-core#250]
Hi! I find that jpf is missing check `static` method, which can be severe! Let me show you an example.
First, you write a class C, which will later be compiled to a method call to use INVOKESTATIC.
```
cat > C.java <<EOF
class C {
   public static void main(String[] a) {
     int i = 123;
     D.m("foobar");
     System.out.println(i);
   }
}
EOF
```
Then you write the static method `m` in class D.
```
cat > D.java <<EOF
class D {
   public static String m(String s) { return s; }
}
EOF
```
Then you can run `javac C.java D.java` to compile, and both `java C` and `java -jar build/RunJPF.jar +classpath=. C` should work fine.

Then you remove that `static` in class D by running `sed -i s/static// D.java; javac D.java`, then `java C` will throw `IncompatibleClassChangeError` as expected.
```
java.lang.IncompatibleClassChangeError: Expected static method D.m(Ljava/lang/String;)Ljava/lang/String;
	at C.main(C.java:4)
```
However, `java -jar build/RunJPF.jar +classpath=. C` will print out `248` (please note that this number should be `123`! ) and end up with `no errors detected`. Missing the check can lead to executing something unexpected and destroy certain variables in the program.
The fix may be adding a check for whether the method to be invoked is actually static or not when executing INVOKESTATIC.
Thanks!

**Comment by cyrille-artho** [comment]
Hi,
Thank you very much for noticing this. This is an issue that we hope won't be too hard to fix, but it may be difficult to test.
We would need the following:
1. C.java and D.java, which are both compiled as usual during gradle's `testCompile` phase.
2. Before the tests run, we'd need a task that (a) creates a modified version of D.java in a temporary directory, (b) compiles it, and (c) moves D.class (the compiled file) to the location of the other compiled files. 
3. Now the tests can run.

This special task in (2) can be implemented using doFirst:
https://docs.gradle.org/current/userguide/build_lifecycle.html

The actions inside that task could be implemented as shell tasks, as in:
https://discuss.gradle.org/t/run-a-process-before-tests/25690/2

However, this would make this particular gradle task less portable and more failure-prone; it should at the very least not prevent the other tests from running if it cannot execute. Perhaps it's also possible to use a gradle-native task to compile the modified D.java.

What do you think?

**Comment by alokprince** [comment]
can I try to solve this issue?
I am a beginner and I am willing to learn and work on this issue as my first issue.

**Comment by cyrille-artho** [comment]
You are welcome to work on this, as I'm not aware of anyone else doing so. As I've written earlier, testing this issue is non-trivial.

**Comment by alokprince** [comment]
I will try my best for this issue.

**Comment by Venkatesh-24** [comment]
Hi I am a beginner. Can I try to solve this issue??

**Comment by cyrille-artho** [comment]
Yes, it seems alokprince is not working on this issue right now, so please give it a try.

**Comment by Venkatesh-24** [comment]
@cyrille-artho can you please point me to related documentation and entry point of the codebase.I am new to this repository.

**Comment by cyrille-artho** [comment]
See the history of this issue and the JPF wiki to get started.
A good way to resolve this issue would be to have a test for it first, which (as described above) is not so simple because code has to be compiled twice.

**Comment by Venkatesh-24** [comment]
Okay I will try my best to solve it.

**Comment by shoyebmd424** [comment]
I can try solve this issue please assign me for this issue

**Comment by CrypticMessenger** [comment]
Hi @cyrille-artho , can I work on this issue? I have identified the changes and tested locally, it resolves the problem, I am currently working on writing Unit test (the way you suggested in this thread).

**Comment by cyrille-artho** [comment]
Yes, there is no recent comment from anyone working on this, so feel free to go ahead.

**Comment by sshaikshoaib** [comment]
Hi I am a beginner. can I work for this issue

**Comment by cyrille-artho** [comment]
Yes, please give it a try. Manually reproducing the existing incorrect behavior is not very hard, but automating this with a special build target is a bit more work. Try one step first and let us know how far you get.

**Comment by siddhesh0705** [comment]
Hi,
I’m Siddhesh Sangale, and I’m new to open-source contributions. I’m eager to get started by tackling some beginner-friendly tasks that can help me build confidence and gain experience. While I’m just beginning my open-source journey, I’m excited to grow my skills and contribute to impactful projects.

I’m looking forward to working on something meaningful and gradually taking on bigger challenges as I learn. I would appreciate any guidance or suggestions for good starting points.

can you appoint me with this issue ?

**Comment by cyrille-artho** [comment]
Yes, a fix for this is fairly easy, and as long as the issue remains open and there has not been any activity for a while, it's safe to assume that it has not been resolved yet.
Feel free to try to fix this in the code, and I can change the issue later to ask for an automated test for this.

**Comment by siddhesh0705** [comment]
@cyrille-artho  can you provide me with the inital steps that how can start with this issue and on which branch I can find this issue ?

**Comment by siddhesh0705** [comment]
@cyrille-artho 
I have submitted a Pull Request to address this issue. Could you please review it and provide feedback or suggest any changes needed to meet your requirements?

**Comment by siddhesh0705** [comment]
Hi,
I hope this message finds you well. I wanted to sincerely apologize for my recent lack of focus on the issue we were working on. My college practical exams have been taking up much of my time and attention, and I regret that I was unable to dedicate the effort required to make meaningful progress.

Now that my exams are behind me, I am committed to refocusing on the problem and catching up. Thank you for your understanding, and I truly appreciate your guidance and patience.

**Comment by 0XKCD0** [comment]
His, can I take this. I'm contributing in Gsoc

**Comment by cyrille-artho** [comment]
Feel free to give it a try. The PR linked to above has a lot of context, so this can help you to see what the problem (and the test) is about.

**Comment by 8packcoder** [comment]
Hey I can solve this issue please assign me this issue.

**Comment by cyrille-artho** [comment]
Feel free to work on it. The issue is currently not being worked on, but PR #510 has much of the needed work already done, so you can use that as a starting point.

**Comment by gauravsingh001-cyber** [comment]
> Hi, Thank you very much for noticing this. This is an issue that we hope won't be too hard to fix, but it may be difficult to test. We would need the following:
> 
> 1. C.java and D.java, which are both compiled as usual during gradle's `testCompile` phase.
> 2. Before the tests run, we'd need a task that (a) creates a modified version of D.java in a temporary directory, (b) compiles it, and (c) moves D.class (the compiled file) to the location of the other compiled files.
> 3. Now the tests can run.
> 
> This special task in (2) can be implemented using doFirst: https://docs.gradle.org/current/userguide/build_lifecycle.html
> 
> The actions inside that task could be implemented as shell tasks, as in: https://discuss.gradle.org/t/run-a-process-before-tests/25690/2
> 
> However, this would make this particular gradle task less portable and more failure-prone; it should at the very least not prevent the other tests from running if it cannot execute. Perhaps it's also possible to use a gradle-native task to compile the modified D.java.
> 
> What do you think?



> Feel free to work on it. The issue is currently not being worked on, but PR [#510](https://github.com/javapathfinder/jpf-core/pull/510) has much of the needed work already done, so you can use that as a starting point.

can i try to solve it..?

**Comment by gauravsingh001-cyber** [comment]
@cyrille-artho 
as a beginner. can you give me chance to work for this issue

**Comment by adishchowdhury** [comment]
Hi @cyrille-artho can I contribute to this issue??
